code/modules/clothes_making/weaving.dm

```diff
@@ -0,0 +1,383 @@
+/*
+
+/obj/structure/spinning_wheel
+/obj/machinery/electric_loom
+
+*/
+
+#define CLOTH_PER_FLAX	2
+
+////////////////////MANUAL LOOM//////////////////////////////////////////////////////////////////////////////////////////
+//TODO: support wool as well, and maybe other plants such as coton
+//I chose linen to start with because of too much time playing Pharaoh
+/obj/structure/spinning_wheel
+	name = "spinning wheel"
+	desc = "Allows you to manually weave grown flax into linen cloth."
+	icon = 'icons/obj/clothes_making.dmi'
+	icon_state = "wooden_loom"
+	density = 1
+	anchored = 0
+	pass_flags_self = PASSMACHINE
+
+	var/remaining_cloth_to_spin = 0
+	var/mob/spinner = null
+
+/obj/structure/spinning_wheel/Destroy()
+	if (spinner)
+		spinner = null
+		processing_objects.Remove(src)
+	..()
+
+/obj/structure/spinning_wheel/examine(mob/user)
+	..()
+	if(remaining_cloth_to_spin > 0)
+		to_chat(user, "<span class='info'>There is enough flax in it to produce [remaining_cloth_to_spin] more cloth sheets.</span>")
```

Comment:
if 1, will say "1 more cloth sheets"; I think there's a macro to handle plurality agreement

Reply:
Not sure the macro handles it unless the word itself is a plural or something, but I'll add something to handle that case.